packages/react-dom/src/client/ReactDOMFiberOption.js

```diff
@@ -36,11 +38,14 @@ function flattenChildren(children) {
 export function validateProps(element: Element, props: Object) {
   // TODO (yungsters): Remove support for `selected` in <option>.
   if (__DEV__) {
-    warning(
-      props.selected == null,
-      'Use the `defaultValue` or `value` props on <select> instead of ' +
-        'setting `selected` on <option>.',
-    );
+    if (!didWarnSelectedSetOnOption) {
```

Comment:
Actually I think this is wrong. This means we only *check* once, not warn once.

Reply:
Fixed in https://github.com/facebook/react/commit/abe0faf3a18ea5f3765f2e0390e4cf0b3f25fc80.

Reply:
Woah good catch!